File: Kotlin/GazaSort.kt
import kotlin.random.Random

fun gazaSort(arr: MutableList<Int>): List<Int> {
    while (!isSorted(arr)) {
        arr.removeAt(Random.nextInt(arr.size))
    }
    return arr
}

fun isSorted(arr: List<Int>): Boolean {
    return arr.zipWithNext().all { (a, b) -> a <= b }
}

File: Scala/GazaSort.scala
import scala.util.Random

def gazaSort(arr: Array[Int]): Array[Int] = {
    while (!isSorted(arr)) {
        val idx = Random.nextInt(arr.length)
        arr.patch(idx, Nil, 1)
    }
    arr
}

def isSorted(arr: Array[Int]): Boolean = {
  (0 until arr.length - 1).forall(i => arr(i) <= arr(i + 1))
}
